### ISSUE TITLE
We have our work cut out for us

### ISSUE THREAD

**Issue by vgturtle127** [piorrro33/psxjin#54]
Alright, I am going to go on a limb here and say that GitHub :octocat:  isn't going to like this repository. :laughing: Here are some reasons:

1. There are many binaries in this repository
2. Some files have weird capitalization and sometimes special characters in them
3. Really, the tests directory can go. Testing shouldn't be handled this way.
4. Some older Visual Studio stuff can probably be updated or deleted
5. We should probably update Lua, libpng, and zlib at least (for security and to fix issues)
6. I should probably go through and use PS1 test homebrew and various BIOS files and see what improvements we can make there before testing the actual games. When we get to the games, I will most likely need an external hard drive, which may require a certain amount of donations in order to get a big enough drive to hold all the data that we would need for good testing.
7. We should check file format compatibility more thoroughly. As in, various types of each file type. An example being ISZ files or CSO files or PBP files.

That's all I can think of now. Does anybody have any ideas about the donation thing? All I would need is a drive big enough so I could format my current machine and have enough data on there to do testing on most of the PS1 library...

**Comment by piorrro33** [comment]
I could help you with testing, though I think you'll have to tell me **the way** I should test them.

**Comment by vgturtle127** [comment]
That would be good. I happen to have a HUGE collection of PS1 games, as I grew up with the system and I also went to a few "haul" type eBay auctions. Anyway, the games and testing isn't really the problem (although I will desperately need your help when we get to game testing, since the PS1 has such a huge library).

![screenshot 20](https://cloud.githubusercontent.com/assets/3183676/8507104/53798768-21f9-11e5-81ff-23ae6a53b5d0.png)

As I speak now, my 1TB hard drive is getting full. Unfortunately, most of it is source code, undumped games, research documents, custom assets, etc. So I can't just delete all of it. :disappointed: 

If I had an external drive I could store data on that and continue working. As of now though, I am kind of stuck. The drives aren't expensive, it's just that I am broke. :disappointed: 

Here is the drive I am looking at:

http://www.amazon.com/Seagate-Expansion-Desktop-External-STEB5000100/dp/B00TKFEEBW/ref=sr_1_1?ie=UTF8&qid=1435997181&sr=8-1&keywords=seagate

If 12 people gave $10, it would be paid for!

And this wouldn't just benefit me, it would benefit everybody because then I could also continue my work on other emulators and software as well.

Anyway, that is the current puzzle I am in. :video_game:

**Comment by piorrro33** [comment]
1. We can't do much for that. 
2. Bad coding practices = more work. 
3. Yup that's useless. 
4. I'll handle that. 
5. I'll handle this as well. 
6. Yes good idea. For the donation thing I'll try to give you a little something :) 
7. I hope that's not too hard to implement. 

But for your hard drive : isn't a 4 TB sufficient? 5 TB is... A lot. Plus the 4 TB costs -20$ so...

**Comment by vgturtle127** [comment]
Actually, we can remove most of the binaries since we aren't using SVN for the repo any more. Unless you use the binaries in the repo to commit or something? I have used SVN before, but I always used TortoiseSVN and I haven't ever used this method. I know FCEUX uses it as well, but they still use SVN and they also aren't on GitHub.

In regards to file formats, the implementation might be difficult depending on what has and hasn't been implemented. However, testing first should be done to see if we even have any work in that regard.

After I thought about it, we probably shouldn't implement support for ISZ or CSO since the PS1 usually doesn't use that, and PBP files won't be necessary since the PSP uses those. Some PS1 emulators support PBP, but honestly if you already played a PS1 game on your PSP, then you probably already know how to convert back and forth.

I have a feeling we could probably rename all the files and have everything still work. Unless the makefiles and Visual Studio solutions are case sensitive, but I have never had this problem because I never use special characters or capitals in my file names. :smile: 

A 4TB would have been sufficient about 6 months ago, but since then I have added much more data to my...collection? And I also have some dead drives that I need to transfer over to my external storage. So, for the future, it would make sense to have plenty of room for my 1TB internal, my 2TB external, my other 1TB external, and my 500GB external. Then I can just trash all the other drives and have one good drive. Hope that wasn't confusing. :laughing:

**Comment by piorrro33** [comment]
In fact, we still use Subversion to retrieve the version that we put in the title bar : we have to keep SubWCRev.exe.
upx.exe packs the executable to get a smaller one : I suppose that we have to keep it.
zip.exe zips the release.
nasmw.exe is the assembler. I don't think it's mandatory, but I'll have to get a closer look at it.

About the file formats, I really have no opinion about them. Feel free to do whatever you want, but I agree about the fact that we shouldn't implement support for ISZ, CSO or PBP.

Visual Studio solutions are not case sensitive neither the Makefiles, AFAIK.

And yeah I understand what you mean. You want to only have one big drive to gather all the others. :smiley:

**Comment by vgturtle127** [comment]
Actually, until the latest version has been tested, I would think we should wait to pack the executable with UPX. I had that problem with software I wrote back when I was on SourceForge. My executables wouldn't run on certain systems, and the packing made the antivirus and malware software go nuts. Just a thought to keep in mind. :smiley: 

I was thinking I could do the zip part as well. Maybe do a .zip release and a .7z release for bandwidth limited connections or something. Just a thought. :smile: 

Do you think I could just rename all the files and then we could fix anything that broke? I don't have enough room on my hard drive for Visual Studio or I would test it myself. :laughing: 

"You want to have one big drive to gather all the others"

Exactly! :smiley:

**Comment by piorrro33** [comment]
In fact, we use UPX and zip through a batch file which is really convenient : you double click on it and everything is ready. So yes, we could do everything manually and remove the binaries but that's less convenient.

**Comment by piorrro33** [comment]
No, I don't think it's useful to make a 7z archive because the zip archive is small enough, IMHO. 
And yeah, feel free to rename all the files, I'll help you to fix them. Ah and BTW, I managed to update libpng (locally, I'll commit it ASAP) but not zlib or Lua. Could you update them for me please? :smiley:

**Comment by vgturtle127** [comment]
Yeah, actually the only file we are "supposed" to need from zlib is zlib.h. I'm not sure if PSXjin uses more than that, but if not we could just keep zlib.h up to date, which would trim down the size of the repo and make updating easy. :smile: 

Lua should be straightforward, but I'm not sure most of the files are needed since most of it is the source to Lua itself. I will check.

I will rename all non-external files first, then start working on libraries.

**Comment by piorrro33** [comment]
In fact, libpng needs zlib to compile and also needs a few other files from 
zlib to work correctly.
For Lua, I had some problems, not because of the update but because of the 
code to modify for it to work with the latest version of Lua.
Yes, feel free to rename them.

**Comment by vgturtle127** [comment]
I renamed all the files, and updated Lua and zlib. Most of the work that needs to be done is Visual Studio-related, or within PSXjin. So that part is done.

**Comment by piorrro33** [comment]
Ah you did a really good work. Thanks! :)

**Comment by vgturtle127** [comment]
Sure, no problem. :sweat_smile:
